checker/tests/initialization/TestPolyInitialized.java

```diff
@@ -0,0 +1,37 @@
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.initialization.qual.NotOnlyInitialized;
+import org.checkerframework.checker.initialization.qual.PolyInitialized;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
+
+public class TestPolyInitialized {
+
+    @NotOnlyInitialized String testStr;
+
+    String test = "test";
+
+    TestPolyInitialized(@UnknownInitialization String str) {
+        // :: error: (method.invocation.invalid)
+        this.testStr = identity(str);
```

Comment:
You should manually compile test cases and ensure the raised error messages are actually what you expect.
Here, I assume you get the `method.invocation.invalid` error because in this constructor `this` is `@UnderInitialization` and the receiver of `identity` requires an `@Initialized` receiver. So to actually test that the assignment is forbidden (or is it allowed?), you need to annotate the receiver parameter of identity and make it `@UnknownInitialization`.

Reply:
The receiver parameter for `identity` isn't annotated. Once that is fixed, try using `NotOnlyInitialized` on the field `testStr`. If that works, we should improve error reporting about using `UnknownInit` on fields.

Reply:
Done